jormungandr/src/settings/logging.rs

```diff
@@ -63,21 +68,31 @@ impl FromStr for LogOutput {
 
 impl LogSettings {
     pub fn to_logger(&self) -> Result<Logger, Error> {
-        let drain = self.output.to_logger(&self.format)?.fuse();
+        let drain = self.output.to_logger(&self.format, &self.backend)?.fuse();
         let drain = slog::LevelFilter::new(drain, self.verbosity).fuse();
         Ok(slog::Logger::root(drain, o!()))
     }
 }
 
 impl LogOutput {
-    fn to_logger(&self, format: &LogFormat) -> Result<Async, Error> {
-        Ok(match self {
-            LogOutput::Stderr => format.decorate_stderr(),
+    fn to_logger(&self, format: &LogFormat, backend: &Option<String>) -> Result<Async, Error> {
+        match self {
+            LogOutput::Stderr => Ok(format.decorate_stderr()),
+            LogOutput::Gelf => 
+                match backend {
+                    Some(graylogHostPort) => {
+                        // We use hostname as source for now, but a better identifier should be used.
+                        let source = get_hostname().unwrap_or("unhostnamed".to_string());
```

Comment:
what about generating a random UUID instead ? So we can protect our users...

Reply:
Did that. But what about using the node id? Would this be too impractical to implement?